Performance.md

```diff
@@ -0,0 +1,290 @@
+Program time:              0.45s
+Bind time:                 0.21s
+Check time:                1.07s
+transformTime time:        0.01s
+commentTime time:          0.00s
+I/O Write time:            0.00s
+printTime time:            0.01s
+Emit time:                 0.01s
+Total time:                1.75s
+```
+
+The most relevant information for most users is:
+
+Field   | Meaning
+--------|---------
+`Files` | the number of files that the program is including (use `listFiles` to see what they are).
+`I/O Read time` | time spent reading from the file system - this includes traversing `include`'d folders.
+`Parse time` | time spent scanning and parsing the program
+`Program time` | combined time spent performing reading from the file system, scanning and parsing the program, and other calculation of the program graph. These steps are intermingled and combined here because files need to be resolved and loaded once they're included via `import`s and `export`s.
+`Bind time` | Time spent building up various semantic information that is local to a single file.
+`Check time` | Time spent type-checking the program.
+`transformTime time` | Time spent rewriting TypeScript ASTs (trees that represent source files) into forms that work in older runtimes.
+`commentTime` | Time spent calculating comments in output files.
+`I/O Write time` | Time spent writing/updating files on disk.
+`printTime` | Time spent calculating the string representation of an output file.
+
+<!-- TODO: is this important? Looks identical to printTime
+
+`Emit time` | ...
+
+-->
+
+## `showConfig`
+
+It's not always obvious what settings a compilation is being run with when running `tsc`, especially given that `tsconfig.json`s can extend other configuration files.
+`showConfig` can explain what `tsc` will calculate for an invocation.
+
+```sh
+tsc --showConfig
+
+# or to select a specific config file...
+
+tsc --showConfig -p tsconfig.json
+```
+
+## `traceResolution`
+
+Running with `traceResolution` can help explain *why* a file was included in a compilation.
+The emit is somewhat verbose, so you might want to redirect output to a file.
+
+```sh
+tsc -p tsconfig.json > resolution.txt
+```
+
+## Running `tsc` Alone
+
+Much of the time, users run into slow performance using 3rd party build tools like Gulp, Rollup, Webpack, etc.
+Running with `tsc --extendedDiagnostics` to find major discrepancies between using TypeScript and the tool can indicate external misconfiguration or inefficiencies.
+
+Some questions to keep in mind:
+
+* Is there a major difference in build times between `tsc` and the build tool with TypeScript integration?
+* If the build tool provides diagnostics, is there a difference between TypeScript's resolution and the build tool's?
+* Does the build tool have *its own configuration* that could be the cause?
+* Does the build tool have configuration *for its TypeScript integration* that could be the cause? (e.g. options for ts-loader?)
+
+## Disabling Editor Plugins
```

Comment:
I would definitely suggest trying straightforward things like this before dumping and reading through an undocumented log format.